torchtitan/checkpoint.py

```diff
@@ -124,9 +124,9 @@ def checkpoint_mp(recv, send):
 class CheckpointManager:
     def __init__(
         self,
-        model: nn.Module,
-        optimizer: torch.optim.Optimizer,
-        lr_scheduler: torch.optim.lr_scheduler.LRScheduler,
+        model_parts: List[nn.Module],
+        optimizers: List[torch.optim.Optimizer],
+        lr_schedulers: List[torch.optim.lr_scheduler.LRScheduler],
```

Comment:
since i changed train.py to always have a list of model parts, we would never pass a single model inside anymore. i could update checkpoint to take a single model in ifs passed, but that codepath would not be exercised.

i could make the wrappers match this style, if you like that better?